CONTRIBUTING.markdown

```diff
@@ -1,114 +1,171 @@
+
+Liferay is known for its innovative top quality features. To maintain this reputation, all code changes are reviewed by a core set of Liferay project maintainers. We encourage you to introduce yourself to the core maintainer(s) and engage them as you contribute to the areas they maintain.
+
+To get a deeper understanding of Liferay in general, make sure to read Liferay’s official documentation on the [Liferay Developer Network](http://dev.liferay.com), for [using](https://dev.liferay.com/discover/portal) and [deploying](https://dev.liferay.com/discover/deployment) Liferay Portal, and [developing](https://dev.liferay.com/develop/tutorials) on Liferay. This documentation contains extensive explanations, examples, and reference material for you to consult time and time again.
+
+For more information, visit the links listed in the Additional Resources section, below.
+
+# Building Liferay Portal
+
+System Requirements:
+* Quad Core i7 Processor
+* 8 GB Memory
+* JDK 8
+* Apache Ant
+* Gradle
+
+The first step to contributing to Liferay Portal is to clone the Liferay Central (**liferay-portal**) repo from Github and build the platform using Apache Ant.  All changes from the Sub Repos are synced to Central so no Sub Repos need to be involved in the build process.  
+
+At the time of this writing the Liferay Central repo weighs in a 12 GB’s.  In order to cut down on the download size the repo can cloned with the options of leaving out commit history and selecting only one branch.
+
+## Build from Central Repo:
+
+To build Liferay Portal from source do the following:
+
+* Fork Liferay Portal Central Repo on Github - [https://github.com/liferay/liferay-portal](https://github.com/liferay/liferay-portal)
+
+* Clone the forked repo’s master branch (or preferred branch) with no commit history:
+
+      git clone https://github.com/<github username>/liferay-portal --branch master --single-branch --depth 1
+
+* Add main liferay-portal repo as an upstream for fetching changes:
+
+      git remote add upstream https://github.com/liferay/liferay-portal
+
+* Build master branch from source for testing customizations:
+
+	    cd liferay-portal
+          ant all
+
+* Test the newly built bundle:
+
+      cd ../bundles
+      ./tomcat-8.0.32/bin/startup.sh
+
+## Building a Module in Central
+
+In order to build a module that is maintained from within Central (aka it isn’t maintained in a Sub Repo as of yet) it will have to be built from within the Central repo.  The good news is that since it’s a module the whole platform does not have to be recompiled.  To build a module located within central do the following:
```

Comment:
I think this is taking for granted that people will know what a Sub Repo is (since it's not explained previously in the article).

Since understanding Sub Repos and how we are using them is hard I think we should simplify things for the potential contributors and make it transparent. To be more specific, I think we should just tell them to work with Central directly, for all modules. That should simplify things for them and it is probably still reasonably easy to move the commits to a subrepo when necessary.

What do you think?